src/bots/filler.ts

```diff
@@ -1957,69 +1962,32 @@ export class FillerBot implements Bot {
 		return nodesSent;
 	}
 
-	protected async tryBulkFillPerpNodesForMarket(
+	protected async tryFillPerpNodesForMarket(
 		nodesToFill: Array<NodeToFill>,
 		buildForBundle: boolean
 	): Promise<number> {
-		const ixs: Array<TransactionInstruction> = [
-			ComputeBudgetProgram.setComputeUnitLimit({
-				units: 1_400_000,
-			}),
-		];
-		if (!buildForBundle) {
-			ixs.push(
-				ComputeBudgetProgram.setComputeUnitPrice({
-					microLamports: Math.floor(
-						this.priorityFeeSubscriber.getCustomStrategyResult()
-					),
-				})
-			);
-		}
-
-		/**
-		 * At all times, the running Tx size is:
-		 * - signatures (compact-u16 array, 64 bytes per elem)
-		 * - message header (3 bytes)
-		 * - affected accounts (compact-u16 array, 32 bytes per elem)
-		 * - previous block hash (32 bytes)
-		 * - message instructions (
-		 * 		- progamIdIdx (1 byte)
-		 * 		- accountsIdx (compact-u16, 1 byte per elem)
-		 *		- instruction data (compact-u16, 1 byte per elem)
-		 */
-		let runningTxSize = 0;
-		let runningCUUsed = 0;
-
-		const uniqueAccounts = new Set<string>();
-		uniqueAccounts.add(this.driftClient.provider.wallet.publicKey.toString()); // fee payer goes first
-
-		const computeBudgetIx = ixs[0];
-		computeBudgetIx.keys.forEach((key) =>
-			uniqueAccounts.add(key.pubkey.toString())
-		);
-		uniqueAccounts.add(computeBudgetIx.programId.toString());
-
-		// initialize the barebones transaction
-		// signatures
-		runningTxSize += this.calcCompactU16EncodedSize(new Array(1), 64);
-		// message header
-		runningTxSize += 3;
-		// accounts
-		runningTxSize += this.calcCompactU16EncodedSize(
-			new Array(uniqueAccounts.size),
-			32
-		);
-		// block hash
-		runningTxSize += 32;
-		runningTxSize += this.calcIxEncodedSize(computeBudgetIx);
-
 		const nodesSent: Array<NodeToFill> = [];
 		let idxUsed = 0;
-		const startingIxsSize = ixs.length;
 		const fillTxId = this.fillTxId++;
+
 		for (const [idx, nodeToFill] of nodesToFill.entries()) {
+			const ixs: Array<TransactionInstruction> = [
+				ComputeBudgetProgram.setComputeUnitLimit({
+					units: 1_400_000,
+				}),
+			];
+			if (!buildForBundle) {
+				ixs.push(
+					ComputeBudgetProgram.setComputeUnitPrice({
+						microLamports: Math.floor(
+							this.priorityFeeSubscriber.getCustomStrategyResult()
+						),
+					})
+				);
+			}
+
 			// do multi maker fills in a separate tx since they're larger
-			if (nodeToFill.makerNodes.length > 1) {
+			if (nodeToFill.makerNodes.length > 2) {
```

Comment:
because we only prepend the pyth ixs if we can fit them in, and right now that size is capped at 2 for maker nodes to include. we don't prepend pyth ixs if we do a multi-maker fill (multi-maker defined as > 2 makers)